lib/berkshelf/packager.rb

```diff
@@ -40,8 +41,18 @@ def initialize(out_file)
     # @return [String]
     #   path to the generated archive
     def run(source)
-      tgz = Zlib::GzipWriter.new(File.open(out_file, "wb"))
-      Archive::Tar::Minitar.pack(Dir.glob("#{source}/*"), tgz)
+      begin
+        dest = Zlib::GzipWriter.open(out_file)
+        tar = RelativeTarWriter.new(dest, source)
+        Find.find(source) do |entry|
+          next if source == entry
+
+          Archive::Tar::Minitar.pack_file(entry, tar)
+        end
+      ensure
+        tar.close
+        dest.close
+      end
```

Comment:
Does it work in a much cleaner fashion if this is instead converted to use ffi-libarchive?

https://github.com/chef/ffi-libarchive

Reply:
I haven't tried it and with apologies I don't have the bandwidth to try a variant solution, but if you were to build one around ffi-libarchive instead of this PR it would not hurt my feelings!

Reply:
using '`ffi-libarchive' ` din't get me right result, so going with current soln.
```
Mixlib::Archive.archive_directory(source, out_file)
 Archive::Reader.open_filename(out_file, nil, strip_components: Dir.pwd.count("/"))
```
(Gives same result)
or

```
tgz = Mixlib::Archive.archive_directory(source, out_file)
Archive::Reader.open_filename(tgz, nil, strip_components: Dir.pwd.count("/"))
```
gives type error